### PR TITLE
change the assemblyArtifact (now libertyRuntime) to Gradle style

### PR DESCRIPTION
**Pull request by dshimo**

### CODE REVIEW

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 1]
```diff
@@ -43,10 +43,11 @@ class Liberty implements Plugin<Project> {
         
         project.extensions.create('liberty', LibertyExtension)
         project.configurations.create('libertyLicense')
+        project.configurations.create('libertyRuntime')
 
         project.task('installLiberty', type: InstallLibertyTask) {
             description 'Installs Liberty from a repository'
-            logging.level = LogLevel.INFO
+            logging.level = LogLevel.DEBUG
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: did you mean to change this to DEBUG?

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy [hunk 2]
```diff
@@ -52,42 +52,18 @@ class InstallLibertyTask extends AbstractTask {
...
-            if (!configNames.contains('InstallLibertyTaskConfig')) {
-                project.getConfigurations().create('InstallLibertyTaskConfig')
-                project.getDependencies().add('InstallLibertyTaskConfig', groupId + ':' +
-                    artifactId + ':' + version)
-            }
+       
+        String runtimeFilePath = project.configurations.getByName('libertyRuntime').getAsPath()
+        if (runtimeFilePath) {
+            logger.debug 'Liberty archive file Path to the local Gradle repository  : ' + runtimeFilePath 
+            println runtimeFilePath
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy]: is this intended to be left in?

File: docs/installLiberty.md [hunk 3]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
...
 
-Note: Only one of the install or assemblyArtifact blocks should be used in the same Gradle project. Use the install block to install from the Liberty repository or from a local file. Use the assemblyArtifact block to install from the Maven repository. If both blocks are specified, the Maven repository with assemblyArtifact is used.
+Note: Use the install block to install from the Liberty repository or from a local file. Use the dependencies block to install from the Maven repository. If both blocks are specified, the dependencies block will take precendence.
+
+### Using the dependencies block
+The Liberty Gradle plugin defines two dependency configurations: `libertyRuntime` and `libertyLicense`.  `libertyRuntime` defines which [WebSphere Liberty runtime](#using-maven-artifact) to download from Maven Central. `libertyLicense` [configures](#license-configuration) a license artifact so that your license JAR archive can be identified and used during the `installLiberty` task. Make sure properly [setup](#installing-your-upgrade-license) your license JAR to prevent a missing dependency failure.
```
Comment (cthigh) [inline docs/installLiberty.md]: Change Maven Central to The Central Repository

File: docs/installLiberty.md [hunk 4]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
 The task can download the WebSphere Liberty runtime archive in three ways: 
 * from a specified location (via `runtimeUrl`)
```
Comment (dshimo) [inline docs/installLiberty.md]: * from a specified location using `runtimeUrl`
done

File: docs/installLiberty.md [hunk 5]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
...
 * from the [Liberty repository](https://developer.ibm.com/wasdev/downloads/) based on a version and a runtime type
-* from the Maven Central repository.  
+* from the [Maven Central repository](http://search.maven.org/) using the dependencies block.
```
Comment (dshimo) [inline docs/installLiberty.md]: done

File: docs/installLiberty.md [hunk 6]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
 The task can download the WebSphere Liberty runtime archive in three ways: 
 * from a specified location (via `runtimeUrl`)
 * from the [Liberty repository](https://developer.ibm.com/wasdev/downloads/) based on a version and a runtime type
-* from the Maven Central repository.  
+* from the [Maven Central repository](http://search.maven.org/) using the dependencies block.  
 
 When installing Liberty from a JAR file, the Liberty license code is needed to install the runtime. When you are installing Liberty from the Liberty repository, you can see the versions of Liberty available to install and find the link to their license using the index.yml file. After opening the license, look for the `D/N: <license code>` line. Otherwise, download the runtime archive and execute `java -jar wlp*runtime.jar --viewLicenseInfo` command and look for the `D/N: <license code>` line.
```
Comment (dshimo) [inline docs/installLiberty.md]: ok

File: docs/installLiberty.md [hunk 7]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
...
 
-Note: Only one of the install or assemblyArtifact blocks should be used in the same Gradle project. Use the install block to install from the Liberty repository or from a local file. Use the assemblyArtifact block to install from the Maven repository. If both blocks are specified, the Maven repository with assemblyArtifact is used.
+Note: Use the install block to install from the Liberty repository or from a local file. Use the dependencies block to install from the Maven repository. If both blocks are specified, the dependencies block will take precendence.
```
Comment (dshimo) [inline docs/installLiberty.md]: done

File: docs/installLiberty.md (hunk 3 above)
Comment (dshimo) [inline docs/installLiberty.md]: 1. done
2. done

File: docs/installLiberty.md [hunk 8]
```diff
@@ -5,11 +5,39 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
...
+
+### Using the dependencies block
+The Liberty Gradle plugin defines two dependency configurations: `libertyRuntime` and `libertyLicense`.  `libertyRuntime` defines which [WebSphere Liberty runtime](#using-maven-artifact) to download from Maven Central. `libertyLicense` [configures](#license-configuration) a license artifact so that your license JAR archive can be identified and used during the `installLiberty` task. Make sure properly [setup](#installing-your-upgrade-license) your license JAR to prevent a missing dependency failure.
+
+You will need to include `group`, `name`, and `version` values for each of your configurations. An `ext` value for a specific archive type can be used but is not required.
```
Comment (dshimo) [inline docs/installLiberty.md]: done

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy (hunk 2 above)
Comment (dshimo) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy]: no, oops

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy (hunk 1 above)
Comment (dshimo) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: yes, i spoke to joseph about this. libertyRun needs to be investigated again

File: docs/installLiberty.md [hunk 9]
```diff
@@ -74,59 +102,14 @@ Use the `install` to specify the name of the Liberty server to install from the
...
-        assemblyArtifact {
-             version = "17.0.0.2"
-        }
-    }    
-  ```
-
 #### Using Maven artifact
 
-Use the `assemblyArtifact` to specify the name of the Maven artifact that contains a custom Liberty server or use one of the provided on the [Maven Central repository](http://search.maven.org/).
+Use the [dependencies block](#using-the-dependencies-block) to specify the name of the Maven artifact that contains a custom Liberty server or use one of the provided on the [Maven Central repository](http://search.maven.org/).
```
Comment (dshimo) [inline docs/installLiberty.md]: done

File: docs/installLiberty.md [hunk 10]
```diff
@@ -3,18 +3,18 @@
 The `installLiberty` task is used to download and install WebSphere Liberty server. The task can also upgrade your Liberty runtime from an ILAN to an IPLA license with a license JAR file [setup](#installing-your-upgrade-license) and [license configuration](#license-configuration).  
 
 The task can download the WebSphere Liberty runtime archive in three ways: 
-* from a specified location (via `runtimeUrl`)
+* from a specified location using `runtimeUrl`
 * from the [Liberty repository](https://developer.ibm.com/wasdev/downloads/) based on a version and a runtime type
-* from the [Maven Central repository](http://search.maven.org/) using the dependencies block.  
+* from the [Central repository](https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/index.yml) using the `libertyRuntime` dependencies configuration.
```
Comment (cthigh) [inline docs/installLiberty.md]: When I look at maven central, they tend to spell it `The Central Repostory`.

That link doesn't look right - not the right link for maven central.  The link to index.yml goes on "index.yml" in the next paragraph.  Sorry if my comment didn't make sense.  There's not a + to add comments on that sentence.

File: docs/installLiberty.md (hunk 6 above)
Comment (cthigh) [inline docs/installLiberty.md]: ....find the link to their license using the [index.yml](https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/index.yml) file.

File: docs/installLiberty.md [hunk 11]
```diff
@@ -117,6 +116,8 @@ The Maven Central repository includes the following Liberty runtime artifacts:
 | [wlp-osgi](https://repo1.maven.org/maven2/com/ibm/websphere/appserver/runtime/wlp-osgi/) | 17.0.0.2, 17.0.0.1, 16.0.0.4, 16.0.0.3, 16.0.0.2, 8.5.5.9, 8.5.5.8 | Liberty runtime with features that support OSGi applications. |
 | [wlp-microProfile1](https://repo1.maven.org/maven2/com/ibm/websphere/appserver/runtime/wlp-microProfile1/) | 17.0.0.2, 17.0.0.1, 16.0.0.4, 16.0.0.3 | Liberty with features for a MicroProfile runtime. |
 
+The `group` value for all the artifacts listed above is `com.ibm.websphere.appserver.runtime`.
```
Comment (cthigh) [inline docs/installLiberty.md]: That works!

File: docs/installLiberty.md [hunk 12]
```diff
@@ -5,9 +5,9 @@ The `installLiberty` task is used to download and install WebSphere Liberty serv
...
+When installing Liberty from a JAR file, the Liberty license code is needed to install the runtime. When you are installing Liberty from the Liberty repository, you can see the versions of Liberty available to install and find the link to their license using the [index.yml](https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/index.yml) file. After opening the license, look for the `D/N: <license code>` line. Otherwise, download the runtime archive and execute `java -jar wlp*runtime.jar --viewLicenseInfo` command and look for the `D/N: <license code>` line.
 
 Note: Use the libertyRuntime dependency to install Liberty from The Central Repository. Use the install block to install from the Liberty repository or from a local file. If both configurations are specified, the libertyRuntime dependency takes precedence.
```
Comment (cthigh) [inline docs/installLiberty.md]: put a tick on all tasks or configuration, such as `libertyRuntime`, `libertyLicense` etc.  Check the whole file.

File: docs/installLiberty.md [hunk 13]
```diff
@@ -57,47 +57,36 @@ Use the `install` to specify the name of the Liberty server to install from the
...
         install {
-            type = "javaee7"
+            runtimeUrl="<url to wlp*.zip>"
         }
     }
   ```
-
-4. Install from a specific location using a zip file.
+  
+3. Install from a specific location.
```
Comment (cthigh) [inline docs/installLiberty.md]: 3. Install from a specific location using a jar file. `licenseCode` is required when installing from a jar file.

**Pull request merged**